services/nih-auth.js

```diff
@@ -4,6 +4,10 @@ const {LOGIN_GOV, NIH} = require("../constants/idp-constants");
 const loginGovRegex = new RegExp(/(?:.){1}(@login.gov){1}\b/i);
 const nihRegex = new RegExp(/(?:.){1}(@nih.gov){1}\b/i);
 
+const validateResponseOrThrow= (res)=> {
+    if (res.status != 200) throw new Error("NIH access token failed to create because of invalid access code or unauthorized access");
```

Comment:
NIH responds with very unclear error message in responce.error; invalid grant.
I included a manual error message instead.